src/renderer/context/socket.context.tsx

```diff
@@ -1,135 +1,221 @@
-import { NodeResult } from "@/renderer/routes/common/types/ResultsType";
-import { SetStateAction, useSetAtom } from "jotai";
-import { createContext, Dispatch, useEffect, useMemo, useState } from "react";
-import { WebSocketServer } from "../web-socket/socket";
-import { v4 as UUID } from "uuid";
-import { SOCKET_URL } from "@/renderer/data/constants";
-import { useHardwareRefetch } from "@/renderer/hooks/useHardwareDevices";
+import { BlockResult } from "@/renderer/types/block-result";
 import {
-  manifestChangedAtom,
-  useFetchManifest,
-  useFetchNodesMetadata,
-} from "@/renderer/hooks/useManifest";
+  createContext,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { v4 as UUID } from "uuid";
 import { toast } from "sonner";
-import { useCustomSections } from "@/renderer/hooks/useCustomBlockManifest";
-import { useSettings } from "@/renderer/hooks/useSettings";
+import { env } from "@/env";
+import { useSettingsStore } from "@/renderer/stores/settings";
+import { useManifestStore } from "@/renderer/stores/manifest";
+import { useShallow } from "zustand/react/shallow";
+import { HTTPError } from "ky";
+import { ZodError } from "zod";
+import { ServerStatus, WorkerJobResponse } from "@/renderer/types/socket";
+import { sendEventToMix } from "@/renderer/services/MixpanelServices";
+import { useSocketStore } from "@/renderer/stores/socket";
+import { useHardwareStore } from "@/renderer/stores/hardware";
 
-type States = {
-  programResults: NodeResult[];
-  setProgramResults: Dispatch<SetStateAction<NodeResult[]>>;
+type SocketState = {
   runningNode: string;
-  serverStatus: IServerStatus;
+  serverStatus: string;
```

Comment:
It's difficult, since there's a bunch of random server statuses that are emitted from the backend that aren't accounted for in the `ServerStatus` enum... I just put string as a quick fix but fixing it properly  will be a bit annoying

Reply:
Fixed